### PR TITLE
`Paywalls`: changed `offerDetails` to be optional

### PR DESCRIPTION
**Pull request by NachoSoto**
In template 4 this is optional, so all templates will support this now.

### CODE REVIEW

File: Sources/Paywalls/PaywallData.swift [hunk 1]
```diff
@@ -81,6 +82,10 @@ extension PaywallData {
             get { return self._callToActionWithIntroOffer }
             set { self._callToActionWithIntroOffer = newValue }
         }
+        public var offerDetails: String? {
```
Comment (NachoSoto) [inline Sources/Paywalls/PaywallData.swift:85]: cc @guido732 this is optional now.

File: RevenueCatUI/Views/IntroEligibilityStateView.swift [hunk 2]
```diff
@@ -38,7 +38,7 @@ struct IntroEligibilityStateView: View {
             // only if there is a custom intro text.
             .withPendingData(self.needsToWaitForIntroEligibility, alignment: self.alignment)
             // Hide if there is no intro but we have no text to ensure layout does not change.
-            .hidden(if: self.isNotEligibleForIntro && self.textWithNoIntroOffer.isEmpty)
+            .hidden(if: self.isNotEligibleForIntro && self.textWithNoIntroOffer == nil)
```
Comment (joshdholtz) [inline RevenueCatUI/Views/IntroEligibilityStateView.swift:41]: Should this test for nil or empty? 🤔

File: RevenueCatUI/Views/IntroEligibilityStateView.swift [hunk 3]
```diff
@@ -49,7 +49,7 @@ struct IntroEligibilityStateView: View {
         } else {
             // Display text with intro offer as a backup to ensure layout does not change
             // when switching states.
-            return self.textWithNoIntroOffer.notEmpty ?? self.textWithIntroOffer ?? ""
+            return self.textWithNoIntroOffer ?? self.textWithIntroOffer ?? ""
```
Comment (joshdholtz) [inline RevenueCatUI/Views/IntroEligibilityStateView.swift:52]: Same thing here with testing for empty still for safety?

**Review by joshdholtz — APPROVED** [review]
Looks good! Just a few questions if we should still be checking for empty string or nil in places 🤷‍♂️

File: RevenueCatUI/Views/IntroEligibilityStateView.swift (hunk 2 above)
Comment (NachoSoto) [reply RevenueCatUI/Views/IntroEligibilityStateView.swift:41]: `PaywallData` now uses `@NonEmptyStringDecodable` for this, so it's guaranteed to not be empty if it's not `nil`.

File: RevenueCatUI/Views/IntroEligibilityStateView.swift (hunk 3 above)
Comment (NachoSoto) [reply RevenueCatUI/Views/IntroEligibilityStateView.swift:52]: Ditto.

**Pull request merged**